### DISCUSSION TITLE
Libre 2 Plus connected for 10 hours then stopped?

### DISCUSSION THREAD

**Discussion by nigho9** [NightscoutFoundation/xDrip#3631] [Q&A] [answered]
I started a new Libre 2 Plus (UK) sensor (this is my first Libre 2 Plus) last night but then found that it failed to receive any readings after 04:44 this morning. Looking at the log screen from after the sensor started, it says "Sensor type unknown, returning Libre1 as failsafe." 
My xdrip version is 645e8b4-2024.04.19 and OOPalgorithm2 was 27da3f5-2023.05.25 until I updated to 2024.05.31 when trying to fix the issue today.
LibreLink works ok.
Scanning with NFC gives various errors eg "NFC read timeout", "NFC I/O error".
Log is now just repeating "We do not have any more recent data to offer than 2024-08-18 04:41:24"
Does anyone have any ideas?

**Comment by parapenT1sta** [comment]
Do you have readings with the official app?

**Reply by nigho9** [reply]
Yes, LibreLink is working perfectly, so sensor is obviously transmitting readings, but xdrip+ seems to be unable to receive them despite having done so for the first 10 or so hours.

**Reply by parapenT1sta** [reply]
Can you share the Status and BT Device pages from System Status?

**Reply by nigho9** [reply]
Sure, here you go:

![Screenshot_20240819-110554](https://github.com/user-attachments/assets/aa7a07fc-6592-457a-a35c-7c88113abec4)
![Screenshot_20240819-110618](https://github.com/user-attachments/assets/8865dcb1-8429-4783-8f6d-7d92c73f6616)

**Reply by parapenT1sta** [reply]
You are not connected to Libre 2, try these settings, at least is working for me with Samsung phones:
https://github.com/NightscoutFoundation/xDrip/discussions/3481#discussioncomment-9594730

**Reply by nigho9** [reply]
Thanks @parapenT1sta but having copied those settings into my xdrip+ there is no difference (btw the phone is Motorola g54)

**Comment by nigelkeirross** [comment]
@tzachi-dar @nigho9 @parapenT1sta On 17 August I tried to start a Libre2 plus (UK) sensor. It failed both "direct", with the initial  xDrip NFC scan (NFC Read Timeout), and also trying it with Bubble Mini bridge. I noticed the logs saying it was unrecognised sensor type, and returned Libre1 as a failsafe.
I edited my local copy of xDrip-plus to initially report the SensorNum & make Libre2 the failsafe result.
The SensorNum for (this) Libre2 plus is 0x6C0931, so I added a case statement for this in LibreOOPAlgorith.java.
With this version I was able to connect the sensor both "direct" (on one phone), and via my Bubble Mini bridge on my backup phone, around 1830 on 17th. Since then, the Bubble Mini connection has been running OK (as at 1241 on 19th ie 42 hours later). 
I did have a very strange incident (more detail in next post) which caused the "direct" readings to stop at 0040 on 18th. I rebooted my phone at 0630, and since then the direct connection has been running "normally" - ie largely connected & getting results, but occasional brief disconnects.

**Reply by nigelkeirross** [reply]
I should also add that I am using the latest OOP2 apk from May 2024. I also downloaded the most recent stable xDrip+ (as indicated from readthedocs - the version from 2024-04-19, and that also failed to recognise/connect to Libre2 plus (UK). My local xDrip+ is based on 2023-05-27 version. I am running on Android 14 on Pixel 5 (Bubble Mini connection) & Pixel8 (direct).

**Reply by Navid200** [reply, marked as answer]
So, you are using a version of xDrip before the recent update that let it recognize the new Libre serial number.

See June 11, 2024 here:  
https://navid200.github.io/xDrip/docs/ReleaseNotes.html

**Reply by nigho9** [reply]
Thanks @Navid200 I just found #2845 which seems to point to the same solution. I will reverse the changes suggested by @parapenT1sta above and install the 11 June version.

**Reply by nigelkeirross** [reply]
@Navid200 Oops! Correct, the most recent version I tried was the most recent stable version according to the read-the-docs page. I was not aware of these Release Notes. Thanks very much, will download that version and try it. Maybe the release notes could mention Libre2 plus explicitly?

**Reply by nigho9** [reply]
Thanks @Navid200 That worked perfectly. And thanks for your input @parapenT1sta. 
Being without xdrip+ for 36 hours and having to rely on LibreLink is a useful reminder of how good xdrip+ is, and how good LibreLink **could** be if Abbot put some effort in! 
Thanks to all the contributors that keep xdrip+ running and improving.

**Reply by nigelkeirross** [reply]
Thanks. I have downloaded it. It does recognise the Libre2 plus, but it does not change the life to 15.5 days (vs usual 14.5). I have also found that it will occasionally crash on NFC.
